File: Contest/3contest/LuckBalance.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int n, k, luck;
    cin >> n >> k;
    int L[n], T[n];
    vector<int> imp;
    for(int i = 0; i < n; ++i){
        cin >> L[i] >> T[i];
    }
    for(int i = 0; i < n; ++i){
        if(T[i] == 1){
            imp.push_back(L[i]);
        }else luck += L[i];
    }
    std::sort(imp.begin(), imp.end());
    for(auto i = imp.end() - 1; i != imp.begin() - 1; --i){
        if(k > 0) {
            luck += *i;
            k--;
        }
        else luck -= *i;
    }
    cout << luck << endl;
}

File: Gym/3contest/LuckBalance.py
s = input()
a = s.find(" ")
n = int(s[0:a])
k = int(s[a+1:len(s)])
L = [0] * n
T = [0] * n
imp = []

luck = 0

for i in range(n):
    s = input()
    a = s.find(" ")
    L[i] = int(s[0:a])
    T[i] = int(s[a+1:len(s)])
    if(T[i] == 1):
        imp.append(L[i])
    else:
        luck += L[i]

imp.sort(reverse = True)

for i in imp:
    if(k > 0):
        luck += i
        k -= 1
    else:
        luck -= i

print(luck)